src/Interpreters/loadMetadata.cpp

```diff
@@ -327,9 +336,21 @@ static void maybeConvertOrdinaryDatabaseToAtomic(ContextMutablePtr context, cons
         }
 
         auto local_context = Context::createCopy(context);
+
+        /// We have to stop background operations that may lock table for share to avoid DEADLOCK_AVOIDED error
+        /// on moving tables from Ordinary database. Server has not started to accept connections yet,
+        /// so there are no user queries, only background operations
+        LOG_INFO(log, "Will stop background operations to be able to rename tables in Ordinary database {}", database_name);
+        for (const auto & action : {ActionLocks::PartsMerge, ActionLocks::PartsFetch, ActionLocks::PartsSend, ActionLocks::DistributedSend})
```

Comment:
you can store the `ActionLocks` in `static constexpr std::array`.

Reply:
Seems like it's not that easy, because it requires to add `const` to forward declarations and it requires to assign some values like `extern const StorageActionBlockType PartsMerge = 1;`, but it will duplicate definitions from `ActionLocksManagers.cpp`...

Reply:
ah I see, it cannot be `constexpr`.
What about `static const std::array`?